src/extension/background-script/actions/lnurl/auth.ts

```diff
@@ -0,0 +1,178 @@
+          origin: message.origin,
+        });
+
+        return { error: e.message };
+      }
+    }
+
+    // if the service returned with a HTTP 200 we still check if the response data is OK
+    if (authResponse?.data.status.toUpperCase() !== "OK") {
+      PubSub.publish(`lnurl.auth.failed`, {
+        authResponse: authResponse,
+        lnurlDetails,
+        origin: message.origin,
+      });
+      return { error: authResponse?.data?.reason };
+    }
+
+    PubSub.publish(`lnurl.auth.success`, {
+      authResponse,
+      lnurlDetails,
+      origin: message.origin,
+    });
+
+    // if auto login should be enabled get the publisher and update the publisher entry
+    if (loginStatus.remember) {
+      allowance = await db.allowances
+        .where("host")
+        .equalsIgnoreCase(message.origin.host)
+        .first();
+      if (allowance?.id) {
+        await db.allowances.update(allowance.id, {
+          lnurlAuth: true,
+        });
+      }
+      await db.saveToStorage();
+    }
+    return { data: authResponse.data };
+  }
+}
+
+/*
+  Execute the LNURL auth
+  returns the response of the LNURL-auth login request
+   or throws an error
+*/
+async function auth(lnurlDetails: LNURLDetails) {
+  if (lnurlDetails.tag !== "login")
+    throw new Error(
+      `LNURL-AUTH FAIL: incorrect tag: ${lnurlDetails.tag} was used`
+    );
+
+  const connector = await state.getState().getConnector();
+
+  // TOOD: the signMessage call can fail
+  const signResponse = await connector.signMessage({
+    message: LNURLAUTH_CANONICAL_PHRASE,
+    key_loc: {
+      key_family: 0,
+      key_index: 0,
+    },
+  });
+  const lnSignature = signResponse.data.signature;
+
+  // make sure we got a signature
+  if (!lnSignature) {
+    throw new Error("Invalid Signature");
+  }
+
+  const hashingKey = sha256(lnSignature).toString(Hex);
+  if (!lnurlDetails.url.host || !hashingKey) {
+    throw new Error("Invalid input");
+  }
+  const linkingKeyPriv = hmacSHA256(lnurlDetails.url.host, hashingKey).toString(
+    Hex
+  );
+  // make sure we got a hashingKey and a linkingkey (just to be sure for whatever reason)
+  if (!hashingKey || !linkingKeyPriv) {
+    throw new Error("Invalid hashingKey/linkingKey");
+  }
+
+  const signer = new HashKeySigner(linkingKeyPriv);
+
+  const k1 = utils.hexToUint8Array(lnurlDetails.k1);
+  if (!lnurlDetails.k1 || !k1) {
+    throw new Error("Invalid K1");
+  }
+  const signedMessage = signer.sign(k1);
+  const signedMessageDERHex = signedMessage.toDER("hex");
+
+  const loginURL = lnurlDetails.url;
+  loginURL.searchParams.set("sig", signedMessageDERHex);
+  loginURL.searchParams.set("key", signer.pkHex);
+  loginURL.searchParams.set("t", Date.now().toString());
+  try {
+    const authResponse = await axios.get<{ status: string; reason?: string }>(
+      loginURL.toString()
+    );
+    return authResponse;
+  } catch (e) {
+    if (axios.isAxiosError(e)) {
+      console.log("LNURL-AUTH FAIL:", e);
```

Comment:
See above. Is this needed? If yes, better avoid `log`. Looks like forgot to remove a debugging statement.